File: Solutions/15. Ransom Note/sol.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        map<char,int> m ;
        for(auto i:magazine)
        {
            m[i]++ ;
        }
        for(auto i:ransomNote)
        {
            if(m[i])
            {
                m[i]-- ;
            }
            else
            {
                return false ;
            }
        }
        return true ;
    }
};

File: Solutions/15. Ransom Note/sol.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        m = collections.defaultdict(int)
        for c in magazine:
            m[c]+=1
        for c in ransomNote:
            if not m[c]:
                return False
            m[c]-=1
        return True